### ISSUE TITLE
loose job data when unhook trailer

### ISSUE THREAD

**Issue by dualznz** [Funbit/ets2-telemetry-server#86]
there seems to be a issue when unhooking the trailer it cancels the information being displayed even though i still have the job.

this shouldn't clear until cancel job or complete job

**Comment by Foshitz** [comment]
I have also noticed this problem for a few days now and I have also noticed that the job info returns after restarting the game. 
I haven't tried just restarting the telemetry server though.

**Comment by dualznz** [comment]
yea i have, it didnt make a difference

**Comment by Funbit** [comment]
If you unhook the trailer within 0.5km of the destination point - the plugin cancels job information. Due to SDK restrictions this is the only way to achieve "job finished" event. 
If you are sure that you are pretty far from the destination when you unhook the trailer - it might be a bug. Please let me know.

**Comment by Foshitz** [comment]
Last time I had that issue was in Leipzig when trucking from Bremen to Linz.

**Comment by dualznz** [comment]
yes i can confirm its a bug i was 1200 km from destination and it cancelled it

**Comment by Funbit** [comment]
Unfortunately, this means that game reports that destination is either 0 or less than 500m (but it shouldn't). So if this is the case - it's more like an SDK problem, rather than the plugin...
Can I ask a question, why do you unhook the trailer in the first place? Where do you place it? o_o

**Comment by dualznz** [comment]
i got stuck so i unhooked so i could move position
and then it happend

**Comment by Funbit** [comment]
I see. 
Well, if you're using the latest version of the server (3.2.5), then we should ask SDK developers, why distance changes along the way to [0, 500m]..

**Comment by dualznz** [comment]
yea not sure, but ok so good old scs bug

**Comment by dualznz** [comment]
maybe if we can store the trailer info, and if it detects it being unhooked it holds the information 
for a min or 2 then kills it if it hasnt been rehooked

**Comment by Foshitz** [comment]
I'm unhooking the trailer in the intersection outside the garage on my hunt for the Property Magnate achievement, so I don't have to drive in and then back up with the trailer to be able to continue my journey

**Comment by dualznz** [comment]
mmmm, i tried 4 jobs so far in both games and they do the same thing, im running the latest

**Comment by Funbit** [comment]
@dualznz Actually, you proposed a very good idea regarding job info buffer. Probably I will implement it. So whenever you unhook your trailer the plugin will clear the job but remember it in some buffer. If trailer is hooked again - restore from the buffer. It should solve the issue.

**Comment by dualznz** [comment]
yea, that would work great indeed

**Comment by dualznz** [comment]
maybe you could add the buffer to the json output showing thats its stored in the buffer with a activation by a boolean True / False, then adding the trailer id and so forth to that container.

this will allow it to still be shown on the API

**Comment by WolfJMZ** [comment]
I also can confirm this bug.
